front/src/applications/operationalStudies/hooks.ts

```diff
@@ -47,21 +47,41 @@ export const useSetupItineraryForTrainUpdate = (
   useEffect(() => {
     const setupItineraryForTrainUpdate = async () => {
       const trainSchedule = await getTrainScheduleById({
-        id: trainScheduleIDsToModify[0],
+        id: trainIdToEdit,
       }).unwrap();
 
       if (infraId) {
         const rollingStock = await getRollingStockByName({
           rollingStockName: trainSchedule.rolling_stock_name,
         }).unwrap();
 
+        const pathItems: PathfindingInputV2['path_items'] = trainSchedule.path.map((step) => {
+          if ('uic' in step) {
+            return { uic: step.uic, secondary_code: step.secondary_code };
+          }
+          if ('track' in step) {
+            return {
+              track: step.track,
+              offset: step.offset,
+            };
+          }
+          if ('operational_point' in step) {
+            return {
+              operational_point: step.operational_point,
+            };
+          }
+          return {
+            trigram: step.trigram,
+          };
+        });
```

Comment:
This block of code is used pathfinding/utils.ts but because we need this to increment pathSteps in store which doesn't have exactly the same properties. But as you pointed, `trainSchedule.path` and `PathfindingInputV2['path_items']` have exactly the same format.

Reply:
This has been removed.